app/src/org/commcare/android/adapters/MenuAdapter.java

```diff
@@ -87,9 +85,7 @@ public MenuAdapter(Context context, CommCarePlatform platform, String menuID){
                         try {
                             XPathExpression mRelevantCondition = m.getCommandRelevance(m.indexOfCommand(command));
                             if(mRelevantCondition != null) {                            
-                                if (ec == null) {
-                                    ec = asw.getEvaluationContext();
-                                }
+                                ec = asw.getEvaluationContext();
```

Comment:
This caching should still be working, btw, since the EC here is global to the session